src/Compression/CompressedReadBufferBase.h

```diff
@@ -38,14 +38,15 @@ class CompressedReadBufferBase
     ///
     /// Returns number of compressed bytes read.
     size_t readCompressedData(size_t & size_decompressed, size_t & size_compressed_without_checksum, bool always_copy);
-
+    size_t readCompressedDataBlockHold(size_t & size_decompressed, size_t & size_compressed_without_checksum);
```

Comment:
Not clear for the reader what "Hold" stands for.

Reply:
First, need understand detailed workflow in function as below:
./ClickHouse/src/Compression/CompressedReadBufferFromFile.cpp
size_t CompressedReadBufferFromFile::readBig(char * to, size_t n)

There are three important buffers inside:
1. "compressed_in" means compressed data streaming buffer for input， defined in CompressedReadBufferBase
2. "to" is the parameter of readbig, it means decompressed buffer for output 
3. "own_compressed_buffer" is the buffer to store compressed block header which is reading from "compressed_in"

The workflow can be simplified as 2 steps:
1. Read block header from "compressed_in" to "own_compressed_buffer". According to information of block header, it can indicate the start address and end address for current block by shifting offset in "compressed_in"。
2. Decompress block data from "compressed_in" to "to" according to block header.

Now the problem is the compressed block may not be completely contained in "compressed_in" buffer which means compressed block may be read across the "compressed_in". In that case, "compressed_in->next" would be called which means working buffer inside would be updated from file streaming.

For native LZ4/ZSTD, it has no probem in facing situation above because they are sychronous. But for QPL deflate asynchronous decompression, it requires source and target buffer for decompression can not be overwritten until execution complete. Hence, we have to define a separate function: readCompressedDataBlockHold to avoid the situation of "read compressed block across the compressed_in".  If readCompressedDataBlockHold return ok, it will call asynchronous API to boost performance , otherwise fackback to sychronous API.

Reply:
How about rename readCompressedDataBlockHold to readCompressedDataBlockForAsynchronous ?

Reply:
![image](https://user-images.githubusercontent.com/19898583/178941274-6721ab18-cdb6-4191-b383-8230c1cf8590.png)

Reply:
Done